### PR TITLE
[MCHECKSTYLE-393] update doxia to 1.9.2

### PR DESCRIPTION
**Pull request by juliendemaziere**
I did a little more than the JIRA https://issues.apache.org/jira/browse/MCHECKSTYLE-393 : I updated all doxia and then fixed dependencies and Java code accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -65,9 +65,9 @@ under the License.
   <properties>
     <!-- Because Checkstyle 7+ requires Java 8 -->
     <javaVersion>8</javaVersion>
-    <mavenVersion>3.0</mavenVersion>
+    <mavenVersion>3.6.3</mavenVersion>
```
Comment (slachiewicz) [inline pom.xml:68]: please do not update the minimum required Maven version to the latest available maven. Currently, our minimum should be 3.1.1. Please also do not update everything in one ticket/PR.

File: pom.xml [hunk 2]
```diff
@@ -234,7 +209,7 @@ under the License.
     <dependency>
       <groupId>org.apache.maven.plugin-testing</groupId>
       <artifactId>maven-plugin-testing-harness</artifactId>
-      <version>2.1</version>
+      <version>3.3.0</version>
```
Comment (slachiewicz) [inline pom.xml:212]: 3.1.0 max

File: pom.xml [hunk 3]
```diff
@@ -243,6 +218,12 @@ under the License.
       <version>${mavenVersion}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.7.30</version>
```
Comment (slachiewicz) [inline pom.xml:224]: 1.7.29 max

**Comment by slachiewicz** [comment]
many CI build failed - please check locally if `mvn verify -Prun-its` run was successfull.

**Comment by juliendemaziere** [comment]
Ok I will fix the versions and run the IT localy.

File: pom.xml (hunk 3 above)
Comment (juliendemaziere) [reply pom.xml]: I removed this dependency as it added many unwanted logs.

File: pom.xml (hunk 2 above)
Comment (juliendemaziere) [reply pom.xml]: Done, where can I find the maximum version(s) not to exceed ?

Comment (slachiewicz) [reply pom.xml]: check https://maven.apache.org/plugin-testing/maven-plugin-testing-harness/dependency-management.html for maven-core dependency. The latest version depends on Maven 3.2.5

File: pom.xml (hunk 1 above)
Comment (juliendemaziere) [reply pom.xml]: Done, I had to add some dependencies (sisu) to make it work.

**Comment by juliendemaziere** [comment]
Thank you for your comments/feedbacks, it is my first PR on an open source project :)

**Comment by slachiewicz** [comment]
Running on our CI platform: https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-checkstyle-plugin/job/MCHECKSTYLE-393v2/

**Comment by slachiewicz** [comment]
I'll split changes here into 2 separate commits - one related to upgrading the minimum required Maven version and to changes related to Doxia. 
small question related to commit message - now It says that the author is "docker host internal" is it expected?

**Comment by jira-importer** [comment]
Resolve #305

**Pull request closed without merging**